pkg/xds/context/destination_index.go

```diff
@@ -0,0 +1,150 @@
+package context
```

Comment:
BTW I think if we want to start making things a bit smoother to use we may want to add [internal packages](https://www.bytesizego.com/blog/golang-internal-package) I don't know if it's the case here but I think this would greatly help us defining more clear API boundaries.

Reply:
I am not sure if DestinationIndex should be moved into internal package. It still exposes public api that is used in other places